### PR TITLE
Add MemoryStorage integration tests to reach all storage impl codes

### PR DESCRIPTION
**Pull request by panarch**

### CODE REVIEW

File: src/storages/memory_storage/mod.rs [hunk 1]
```diff
@@ -38,8 +38,7 @@ impl Store<Key> for MemoryStorage {
     async fn fetch_schema(&self, table_name: &str) -> Result<Option<Schema>> {
         self.items
             .get(table_name)
-            .map(|item| item.schema.clone())
-            .map(Ok)
+            .map(|item| Ok(item.schema.clone()))
```
Comment (ever0de) [inline src/storages/memory_storage/mod.rs:41]: Q: Is there any reason to change this?

File: tests/memory_storage.rs [hunk 2]
```diff
@@ -24,3 +24,98 @@ impl Tester<Key, MemoryStorage> for MemoryTester {
...
+            .as_ref()
+            .map(Vec::len),
+        Ok(0),
+    );
+
+    assert_eq!(
+        block_on(storage.scan_indexed_data("Idx", "hello", None, None)).map(|_| ()),
+        Err(Error::StorageMsg(
+            "[MemoryStorage] index is not supported".to_owned()
+        ))
```
Comment (ever0de) [inline tests/memory_storage.rs:93]:
```suggestion
        block_on(storage.scan_indexed_data("Idx", "hello", None, None)).unwrap_err(),
        Error::StorageMsg(
            "[MemoryStorage] index is not supported".to_owned()
        )
```
how about this change?

Comment (ever0de) [reply tests/memory_storage.rs:93]: If you change it, you have to change the other conditions as well.

**Review by ever0de — COMMENTED** [review]
Minor comment (nitpicking)

File: src/storages/memory_storage/mod.rs (hunk 1 above)
Comment (panarch) [reply src/storages/memory_storage/mod.rs:41]: Nothing special reason, `cargo-tarpaulin` marks two `map` as not reached codes, so I just merged two independent lines into an one.

File: tests/memory_storage.rs (hunk 2 above)
Comment (panarch) [reply tests/memory_storage.rs:93]: The content `Ok` of `storage.scan_indexed_data` is `Iterator` and it does not implement `Debug`.
`unwrap_err` requires `Debug` impl for the `Ok(T)`, so I simply replaced `Iterator` to `()`.

Comment (ever0de) [reply tests/memory_storage.rs:93]: oh, need a debug implementation
It's good to leave it like this.

**Review by ever0de — APPROVED** [review]
No other changes seem necessary

**Pull request merged**